src/components/events/Calendar.tsx

```diff
@@ -0,0 +1,145 @@
+
+      const today = new Date();
+      const timeMin = new Date(
+        today.getTime() - 60 * 60 * 24 * 7 * 5 * 1000,
+      ).toISOString();
+      const timeMax = new Date(
+        today.getTime() + 60 * 60 * 24 * 7 * 5 * 1000,
+      ).toISOString();
+
+      const url = `https://www.googleapis.com/calendar/v3/calendars/${calendarId}/events?key=${apiKey}&orderBy=startTime&singleEvents=true&timeMin=${encodeURIComponent(
+        timeMin,
+      )}&timeMax=${encodeURIComponent(timeMax)}`;
+
+      try {
+        const response = await fetch(url);
+        if (!response.ok) {
+          throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        const data = await response.json();
+        const events = data.items.map(
+          ({
+            start,
+            end,
+            location,
+            description,
+            summary,
+          }: GoogleEventProps) => ({
+            start: start.dateTime,
+            end: end.dateTime,
+            location,
+            description,
+            title: summary,
+          }),
+        );
+        return events;
+      } catch (error) {
+        console.error("Error fetching events from Google Calendar:", error);
+      }
+    },
+  });
+
+  return (
+    <>
+      {
+        <Dialog
+          open={Object.keys(current).length > 0}
+          onOpenChange={() => setCurrent({})}
+        >
+          <DialogContent>
+            <DialogHeader>
+              <DialogTitle>
+                <p className="text-xl">{current.title}</p>
+                <p className="text-base font-normal">
+                  {current.location} from{" "}
+                  {new Date(current.start as string).toLocaleTimeString(
+                    "en-US",
+                    {
+                      hour: "2-digit",
+                      minute: "2-digit",
+                    },
+                  )}{" "}
+                  to{" "}
+                  {new Date(current.end as string).toLocaleTimeString("en-US", {
+                    hour: "2-digit",
+                    minute: "2-digit",
+                  })}
+                </p>
+              </DialogTitle>
+              <DialogDescription>{current.description}</DialogDescription>
+            </DialogHeader>
+          </DialogContent>
+        </Dialog>
+      }
+
+      <Calendar
+        mode="single"
+        selected={new Date()}
+        className="w-5/6 sm:w-3/4"
+        events={data}
+        setCurrent={setCurrent}
+      />
+
+      <UpcomingTitle title="Upcoming Events" />
+      <div className="flex flex-col items-center space-y-4">
+        {data
+          ?.filter((event: GoogleEventProps) => {
+            const eventDate = new Date(event.start.toString());
+            const currentDate = new Date();
+            const range = new Date();
+            range.setDate(currentDate.getDate() + 7);
+            return eventDate >= currentDate && eventDate <= range;
+          })
```

Comment:
```suggestion
          })
          .slice(0, 3)
```

only show three closest events for a cleaner look